pkg/pillar/cmd/zedrouter/dnsmasq.go

```diff
@@ -206,6 +206,13 @@ func createDnsmasqConfiglet(
 		// dom0 virtual router that makes the packets pass through ACLs and flow log.
 		file.WriteString(fmt.Sprintf("dhcp-option=option:netmask,%s\n",
 			"255.255.255.255"))
+		if advertizeRouter {
```

Comment:
Yeah, I think this might be needed when the app has multiple ports connected to network. Let me play around a bit with this change.

Reply:
I have tested with a similar change for container/vm running on hypervisor and the behavior is not the same. I do not see these additional routes getting added.
In the case of container, it is because the pkg/xen-tools/initrd/udhcpc_script.sh does not add classless static routes.
And for the VM case, the VM needs to know how the garteway can be reached. It is either or between routers and classless static routes.

```+                       file.WriteString(fmt.Sprintf("dhcp-option=option:classless-static-route,%s/32,%s,%s,%s,%s,%s\n",
+                               router, "0.0.0.0",
+                               "0.0.0.0/0", router,
+                               netconf.Subnet.String(), router))```

With the above change, VM networking worked. But, still now completely when we have multiple interfaces connected to internet.

Reply:
Good point, I'll try this option.
I am in the process of fixing this issue in #1327 . 
As for the VM, apparently it depends on the dhcp client. Which client is the problem with?

Reply:
@giggsoff https://github.com/lf-edge/eve/pull/1337
I have put together all fixes in this PR. Please have a look and let me know.